internal/nodes/validator.go

```diff
@@ -517,86 +532,88 @@ func (v *Validator) ParseHostPathForRoute(ns string, routeName string, routeSpec
+					// for sni VS created using hostrule
+					additionalTLS = TlsSettings{
+						SecretName: secretName,
+						Hosts:      hostMap,
+					}
 				}
-				tls.cacert = routeSpec.TLS.CACertificate
-				if routeSpec.TLS.InsecureEdgeTerminationPolicy == routev1.InsecureEdgeTerminationPolicyRedirect {
-					tls.redirect = true
-				} else if routeSpec.TLS.InsecureEdgeTerminationPolicy == routev1.InsecureEdgeTerminationPolicyNone {
-					tls.blockHTTPTraffic = true
+				tlsConfigs = append(tlsConfigs, additionalTLS)
+			} else if routeSpec.TLS != nil {
+				// build edge cert data for termination: edge and reencrypt
+				if routeSpec.TLS.Termination == routev1.TLSTerminationEdge ||
+					routeSpec.TLS.Termination == routev1.TLSTerminationReencrypt {
+					if routeSpec.TLS.Certificate == "" || routeSpec.TLS.Key == "" {
+						secretName = lib.GetDefaultSecretForRoutes()
+						tls.SecretName = secretName
+						tls.SecretNS = utils.GetAKONamespace()
+					} else {
+						tls.cert = routeSpec.TLS.Certificate
+						tls.key = routeSpec.TLS.Key
+					}
+					tls.cacert = routeSpec.TLS.CACertificate
+					if routeSpec.TLS.InsecureEdgeTerminationPolicy == routev1.InsecureEdgeTerminationPolicyRedirect {
+						tls.redirect = true
+					} else if routeSpec.TLS.InsecureEdgeTerminationPolicy == routev1.InsecureEdgeTerminationPolicyNone {
+						tls.blockHTTPTraffic = true
+					}
 				}
-			}
 
-			// reencrypt specific
-			if routeSpec.TLS.Termination == routev1.TLSTerminationReencrypt {
-				tls.reencrypt = true
-				if routeSpec.TLS.DestinationCACertificate != "" {
-					tls.destCA = routeSpec.TLS.DestinationCACertificate
-				}
+				// reencrypt specific
+				if routeSpec.TLS.Termination == routev1.TLSTerminationReencrypt {
+					tls.reencrypt = true
+					if routeSpec.TLS.DestinationCACertificate != "" {
+						tls.destCA = routeSpec.TLS.DestinationCACertificate
+					}
 
-				// overwrite with httprule
-				if useHttpRuleCA, caCert := destinationCAHTTPRulePresent(key, hostName, routeSpec.Path); useHttpRuleCA {
-					tls.destCA = caCert
+					// overwrite with httprule
+					if useHttpRuleCA, caCert := destinationCAHTTPRulePresent(key, hostName, routeSpec.Path); useHttpRuleCA {
+						tls.destCA = caCert
+					}
 				}
+				tlsConfigs = append(tlsConfigs, tls)
 			}
-			tlsConfigs = append(tlsConfigs, tls)
-		}
 
-		ingressConfig.TlsCollection = tlsConfigs
-		if !useHostRuleSSL {
-			// If svc for a route gets processed before the route itself,
-			// then secret mapping may not be updated, update it here.
-			if ok, _ := objects.OshiftRouteSvcLister().IngressMappings(ns).GetIngToSecret(routeName); !ok {
-				akoNS := utils.GetAKONamespace()
-				objects.OshiftRouteSvcLister().IngressMappings(ns).AddIngressToSecretsMappings(akoNS, routeName, secretName)
-				objects.OshiftRouteSvcLister().IngressMappings(akoNS).AddSecretsToIngressMappings(ns, routeName, secretName)
+			ingressConfig.TlsCollection = tlsConfigs
```

Comment:
@aaha97 : We can set this variable after for loop for secret.

Reply:
moved the variable update